source/server/admin/config_dump_handler.cc

```diff
@@ -133,15 +134,22 @@ bool shouldIncludeEdsInDump(const Http::Utility::QueryParams& params) {
   return Utility::queryParam(params, "include_eds") != absl::nullopt;
 }
 
-Matchers::StringMatcherPtr buildNameMatcher(const Http::Utility::QueryParams& params) {
+absl::StatusOr<Matchers::StringMatcherPtr>
+buildNameMatcher(const Http::Utility::QueryParams& params) {
   const auto name_regex = Utility::queryParam(params, "name_regex");
   if (!name_regex.has_value()) {
     return std::make_unique<Matchers::UniversalStringMatcher>();
   }
   envoy::type::matcher::v3::RegexMatcher matcher;
   *matcher.mutable_google_re2() = envoy::type::matcher::v3::RegexMatcher::GoogleRE2();
   matcher.set_regex(*name_regex);
+  TRY_ASSERT_MAIN_THREAD
   return Regex::Utility::parseRegex(matcher);
```

Comment:
Ultimately it would be nice to have a version of this that doesn't throw given this very short call stack, but not a blocker for this PR.